lib/service/contract_service.dart

```diff
@@ -8,7 +8,8 @@ typedef TransferEvent = void Function(
 );
 
 abstract class IContractService {
-  Future<Credentials> getCredentials(String privateKey);
+  EthPrivateKey  getCredentials(String privateKey) => EthPrivateKey.fromHex(privateKey);
```

Comment:
this shouldn't need implementation. consider having the method signature.
`EthPrivateKey getCredentials(String privateKey);`